streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala

```diff
@@ -100,7 +100,7 @@ class Subscription(
   def sendMessage(msg: Message, partition: Int): Int = {
 
     // only send message whose timestamp matches the lifeTime
-    if (life.contains(msg.timestamp)) {
+    if (partition >= 0 && life.contains(msg.timestamp)) {
```

Comment:
Where you return partition = -1? 
Can you make that explicit so that we don't accidentially delete the check in future refactory?